### PR TITLE
common: create custom chain `common` from geth genesis file

### PR DESCRIPTION
**Pull request by acolytec3**
Adds a new static constructor to `common` that allows the creation of a `common` instance from genesis file that follows the [geth/puppeth format](https://geth.ethereum.org/docs/interface/private-network)

### CODE REVIEW

File: packages/common/scripts/gethGenesisConverter.ts
Comment (ryanio) [inline packages/common/scripts/gethGenesisConverter.ts:2]:
```suggestion
import { BlockHeader } from '@ethereumjs/block'
```

File: packages/common/src/index.ts
Comment (ryanio) [inline packages/common/src/index.ts:311]:
```suggestion
   * There is a helper script `scripts/gethGenesisConverter.ts` that takes the genesis file [layout](https://geth.ethereum.org/docs/interface/private-network)
```

File: packages/common/src/index.ts
Comment (ryanio) [inline packages/common/src/index.ts:323]:
```suggestion
   * Using with {@link Common.fromCustomGenesis}
```

File: packages/common/scripts/gethGenesisConverter.ts
Comment (holgerd77) [inline packages/common/scripts/gethGenesisConverter.ts:2]: wrong import

File: packages/common/src/index.ts
Comment (holgerd77) [inline packages/common/src/index.ts:339]: Ah, no, I am not going along with this. That makes not so much sense actually to more or less replicate the Common custom chain format constructor. The main logic of parsing and everything should be in this constructor, otherwise there is not much gained.

So this should take in the geth genesis.json file.

Comment (holgerd77) [reply packages/common/src/index.ts:339]: Or am I missing something, maybe I am also a bit confused. Sorry, need to go to bed with my kids, please don't merge this PR yet. Would want to do another review.

**Comment by holgerd77** [comment]
Rebased this.

File: packages/client/bin/cli.ts
Comment (holgerd77) [inline packages/client/bin/cli.ts:196]: This `params` naming is obviously just some relic from the past. But we can keep for this PR, will update when I do the Common chain file format integration.

Some other note: the whole naming of methods in `util/parse.ts` is not so optimal (not introduced by this PR but rather taken from the previous work). Some methods are labelled `parse*` and are doing something completely different. Geth-specific functionality is sometimes labelled accordingly, sometimes not. This should be updated at some point (especially the two exported methods `parseParams` and `parseGenesisState` should get some Geth prefix (`parseGethParams`).

**Review by holgerd77 — APPROVED** [review]
LGTM, cool, thanks! 😄

**Pull request merged**
